shared/actions/autoreset.tsx

```diff
@@ -1,3 +1,35 @@
-function* autoresetSaga() {}
+import * as Saga from '../util/saga'
+import * as Container from '../util/container'
+
+import * as NotificationsGen from './notifications-gen'
+import * as RouteTreeGen from './route-tree-gen'
+import * as AutoresetGen from './autoreset-gen'
+import * as RPCGen from '../constants/types/rpc-gen'
+
+import * as Constants from '../constants/autoreset'
+
+const receivedBadgeState = async (
+  state: Container.TypedState,
+  action: NotificationsGen.ReceivedBadgeStatePayload
+) => {
+  const newResetState = action.payload.badgeState.resetState
+  if (state.autoreset !== newResetState) {
```

Comment:
This will fail every time, `newResetState` is a new object. I'd say compare the fields individually.